File: app/src/main/java/me/pwcong/histogramdemo/MainActivity.java
package me.pwcong.histogramdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import java.util.ArrayList;

import me.pwcong.histogram.model.HistogramEntry;
import me.pwcong.histogram.view.HistogramView;

public class MainActivity extends AppCompatActivity {

    HistogramView view;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        view= (HistogramView) findViewById(R.id.view);
        view.setData(getEntries());


    }

    private ArrayList<HistogramEntry> getEntries(){

        ArrayList<HistogramEntry> entries=new ArrayList<>();

        entries.add(new HistogramEntry("小明",18));
        entries.add(new HistogramEntry("小红",9));
        entries.add(new HistogramEntry("小黑",40));
        entries.add(new HistogramEntry("小花",25));
        entries.add(new HistogramEntry("小黄",33));

        return entries;

    }

}

File: app/src/main/java/com/github/pwcong/histogramviewdemo/MainActivity.kt
package com.github.pwcong.histogramviewdemo

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.github.pwcong.histogramview.entry.HistogramEntry
import com.github.pwcong.histogramview.view.HistogramView


class MainActivity : AppCompatActivity() {
    private var view: HistogramView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        view = findViewById<View>(R.id.view) as HistogramView
        view!!.setData(entries)
    }

    private val entries: List<HistogramEntry>
        get() {
            val entries = mutableListOf<HistogramEntry>()

            entries.add(HistogramEntry("小明", 18f))
            entries.add(HistogramEntry("小红", 9f))
            entries.add(HistogramEntry("小黑", 40f))
            entries.add(HistogramEntry("小花", 25f))
            entries.add(HistogramEntry("小黄", 33f))

            return entries
        }
}
